src/main/java/io/fabric8/jenkins/openshiftsync/BuildSCMBinder.java

```diff
@@ -0,0 +1,88 @@
+import org.jenkinsci.plugins.workflow.cps.CpsScmFlowDefinition;
+import org.jenkinsci.plugins.workflow.flow.FlowDefinition;
+import org.jenkinsci.plugins.workflow.flow.FlowDefinitionDescriptor;
+import org.jenkinsci.plugins.workflow.flow.FlowExecution;
+import org.jenkinsci.plugins.workflow.flow.FlowExecutionOwner;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.plugins.workflow.multibranch.BranchJobProperty;
+import org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject;
+
+public class BuildSCMBinder extends FlowDefinition {
+
+  public static final String DEFAULT_JENKINS_FILEPATH = "Jenkinsfile";
+  private String jenkinsFilePath;
+
+  BuildSCMBinder(String jenkinsFilePath) {
+    this.jenkinsFilePath = jenkinsFilePath;
+  }
+
+  public FlowExecution create(FlowExecutionOwner handle, TaskListener listener, List<? extends Action> actions) throws Exception {
+    Executable exec = handle.getExecutable();
+
+    if (jenkinsFilePath.isEmpty() || jenkinsFilePath == null){
+      this.jenkinsFilePath = DEFAULT_JENKINS_FILEPATH;
+    }
+
+    if (!(exec instanceof WorkflowRun)) {
+      throw new IllegalStateException("inappropriate context");
+    } else {
+      WorkflowRun build = (WorkflowRun)exec;
+      WorkflowJob job = (WorkflowJob)build.getParent();
+      BranchJobProperty property = (BranchJobProperty)job.getProperty(BranchJobProperty.class);
+      if (property == null) {
+        throw new IllegalStateException("inappropriate context");
+      } else {
+        Branch branch = property.getBranch();
+        ItemGroup<?> parent = job.getParent();
+        if (!(parent instanceof WorkflowMultiBranchProject)) {
+          throw new IllegalStateException("inappropriate context");
+        } else {
+          SCMSource scmSource = ((WorkflowMultiBranchProject)parent).getSCMSource(branch.getSourceId());
+          if (scmSource == null) {
+            throw new IllegalStateException(branch.getSourceId() + " not found");
+          } else {
+            SCMHead head = branch.getHead();
+            SCMRevision tip = scmSource.fetch(head, listener);
+            SCM scm;
+            if (tip != null) {
+              scm = scmSource.build(head, scmSource.getTrustedRevision(tip, listener));
+              build.addAction(new SCMRevisionAction(tip));
+            } else {
+              listener.error("Could not determine exact tip revision of " + branch.getName() + "; falling back to nondeterministic checkout");
+              scm = branch.getScm();
+            }
+
+            return (new CpsScmFlowDefinition(scm, jenkinsFilePath)).create(handle, listener, actions);
+          }
+        }
+      }
+    }
+  }
+
+
+  @Extension
+  public static class DescriptorImpl extends FlowDefinitionDescriptor {
+    public DescriptorImpl() {
+    }
+
+    public String getDisplayName() {
+      return "Pipeline script from Jenkinsfile (Openshift)";
```

Comment:
I'll be more precise this time.  Make it:

`Pipeline script from an OpenShift Pipeline Strategy Build`